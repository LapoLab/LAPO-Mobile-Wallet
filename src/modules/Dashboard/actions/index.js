// @flow
import { createActions } from 'redux-actions';

export { setSendAddress } from '../../Send/actions';

export const {
  getInfoAboutWallet, setIsReady, setAnalitycLoadBlock, closeApp,
} = createActions({
  GET_INFO_ABOUT_WALLET: (data = '') => data,
  SET_IS_READY: (data = true) => data,
  SET_ANALITYC_LOAD_BLOCK: (data = true) => data,
  CLOSE_APP: (data = true) => data,
});

export { getContacts } from '../../ChooseContact/actions';
