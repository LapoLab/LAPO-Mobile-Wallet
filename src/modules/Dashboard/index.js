import Notify from './components';

export default Notify;
