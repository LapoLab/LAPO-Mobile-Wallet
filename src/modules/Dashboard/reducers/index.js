import { handleActions } from 'redux-actions';
import {
  setIsReady,
  setAnalitycLoadBlock,
  getInfoAboutWallet,
} from '../actions';

export const initialDashboardState = {
  isReady: false,
  countBlock: 0,
  maxBlock: 0,
  isStartGetInfo: false,
};

// reducer
const reducer = handleActions(
  {
    [setIsReady](state, { payload }) {
      return {
        ...state,
        isReady: payload,
      };
    },
    [getInfoAboutWallet](state) {
      return {
        ...state,
        isStartGetInfo: true,
      };
    },
    [setAnalitycLoadBlock](state, { payload }) {
      return {
        ...state,
        countBlock: payload.countBlock,
        maxBlock: payload.maxBlock,
      };
    },

  },
  initialDashboardState,
);

export default reducer;
