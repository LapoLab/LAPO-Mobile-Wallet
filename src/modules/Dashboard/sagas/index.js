import getInfoWithPeriod from './getInfoWithPeriod';
import closeApp from './closeApp';

export * from '../actions';

export default {
  getInfoWithPeriod,
  closeApp,
};
