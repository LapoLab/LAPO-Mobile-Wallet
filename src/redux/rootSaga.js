import { all } from 'redux-saga/effects';
import commonsSagas from '../commons/sagas';
import recoveryWordsSagas from '../modules/RecoveryWords/sagas';
import sendSagas from '../modules/Send/sagas';
import paperWordsConfirm from '../modules/PaperWordsConfirm/sagas';
import dashboard from '../modules/Dashboard/sagas';
import settings from '../modules/Settings/sagas';
import paperWords from '../modules/PaperWords/sagas';
import transactionsDetails from '../modules/TransactionDetails/sagas';
import chooseContact from '../modules/ChooseContact/sagas';
import splashScreen from '../modules/SplashScreen/sagas';
import switchContacts from '../modules/SwitchContacts/sagas';
import receive from '../modules/Receive/sagas';

// eslint-disable-next-line
const getListeners = (...args) => args.reduce((acc, nextArg) => [...acc, ...Object.values(nextArg).map(func => func())], []);

export default function* rootSaga() {
  yield all(
    getListeners(
      commonsSagas,
      recoveryWordsSagas,
      sendSagas,
      paperWordsConfirm,
      dashboard,
      settings,
      paperWords,
      transactionsDetails,
      chooseContact,
      splashScreen,
      switchContacts,
      receive,
    ),
  );
}
