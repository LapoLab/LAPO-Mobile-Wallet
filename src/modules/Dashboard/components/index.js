/**
 * @flow
 */

import React, { Component, Fragment } from 'react';
import {
  Text,
  View,
  TouchableOpacity,
  Image,
  SafeAreaView,
  BackHandler,
} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import { NavigationActions } from 'react-navigation';
import { connect } from 'react-redux';

import { bindActionCreators } from 'redux';
import {
  setModalSend as setModalSendAction,
  setModalReceive as setModalReceiveAction,
  setBalance as setBalanceAction,
  setAddress as setAddressAction,
  setTransactions as setTransactionsAction,
  setModalTxDetails as setModalTxDetailsAction,
  closeApp as closeAppAction,
} from '../../../commons/actions';
import SpinnerPage from '../../../commons/components/SpinnerPage';
import * as actionNotify from '../../Notify/actions';
import * as actionAlert from '../../Alert/actions';
import * as actionDashboard from '../actions';

import Modals from './Modals';

import Balance from './Balance';
import Transactions from './Transactions';
import Notify from '../../Notify/component';
import styles from './styles';
import {
  menuIcon,
  ActivityIcon,
  iconNotifyActive,
  dashboardIcon,
  iconSend,
  iconRecieve,
} from '../../../assets';

type Props = {
  isModalSend: boolean,
  isModalReceive: boolean,
  txDetails: string,
  navigation: {
    openDrawer: Function,
    closeDrawer: Function,
    dispatch: Function,
    navigate: Function,
  },
  setModalSend: Function,
  getContacts: Function,
  setModalReceive: Function,
  showAlert: Function,
  transactions: [Object],
  setVisibleNotify: Function,
  isVisibleNotify: boolean,
  timeNotifyLast: number,
  getInfo: Function,
  closeApp: Function,
  isStartGetInfo: boolean,
  isOpenDrawer: boolean,
  isModalSwitchContacts: boolean,
};

class Dashboard extends Component<Props> {
  componentDidMount() {
    const { getInfo, isStartGetInfo, getContacts } = this.props;

    getContacts();

    if (!isStartGetInfo) getInfo();
    BackHandler.addEventListener('hardwareBackPress', this.handleBack);
  }

  componentWillUnmount() {
    BackHandler.removeEventListener('hardwareBackPress', this.handleBack);
  }

  handleBack = async () => {
    const {
      closeApp,
      isOpenDrawer,
      showAlert,
      navigation: { closeDrawer },
    } = this.props;

    if (isOpenDrawer) {
      closeDrawer();
      return true;
    }

    // showAlert();
    showAlert(
      'Confirm Exit',
      'Do you really want to exit the app?',
      // eslint-disable-next-line
      [{ text: 'OK', onPress: () => closeApp() }, { text: 'CANCEL' }],
      {
        cancelable: true,
      },
    );
    return true;
  };

  handleShowMenu = () => {
    const {
      navigation: { openDrawer },
    } = this.props;
    openDrawer();
  };

  navigateToScreen = route => () => {
    const {
      navigation: { dispatch },
    } = this.props;
    const navigateAction = NavigationActions.navigate({
      routeName: route,
    });
    dispatch(navigateAction);
  };

  handleSend = () => {
    const { setModalSend } = this.props;
    setModalSend(true);
  };

  handleRecieve = async () => {
    const { setModalReceive } = this.props;
    setModalReceive(true);
  };

  handleShowNotify = () => {
    const { setVisibleNotify, isVisibleNotify } = this.props;
    setVisibleNotify(!isVisibleNotify);
  };

  render() {
    const {
      navigation,
      isVisibleNotify,
      transactions,
      timeNotifyLast,
      isModalSend,
      isModalReceive,
      txDetails,
      isModalSwitchContacts,
    } = this.props;

    const txsWithoutOut = transactions.filter(item => item.mode === 'out');

    const isNotifyActive = !!(
      txsWithoutOut.length > 0 && txsWithoutOut[txsWithoutOut.length - 1].time > timeNotifyLast
    );

    const modalVisible = isModalSend || isModalReceive || (txDetails !== '' && isModalSwitchContacts !== '');

    return (
      <Fragment>
        <LinearGradient
          start={{ x: -0.1, y: -0.1 }}
          end={{ x: 1.2, y: 1.2 }}
          locations={[0, 1]}
          colors={['#4183BD', '#093B0C']}
          style={{ flex: 1, width: '100%' }}
        >
          <SafeAreaView style={styles.container}>
            <View style={styles.header}>
              <TouchableOpacity style={styles.buttonMenu} onPress={this.handleShowMenu}>
                <Image style={styles.menuHeader} source={menuIcon} />
              </TouchableOpacity>

              <Image source={dashboardIcon} style={styles.dashboardIcon} />

              <TouchableOpacity style={styles.buttonNotify} onPress={this.handleShowNotify}>
                <Image
                  style={styles.notifyHeader}
                  source={isNotifyActive ? iconNotifyActive : ActivityIcon}
                />
              </TouchableOpacity>
            </View>

            <Balance />

            <View style={styles.body}>
              <View style={styles.buttonGroup}>
                <TouchableOpacity style={styles.buttonLeft} onPress={this.handleSend}>
                  <Image style={styles.iconSend} source={iconSend} />
                  <Text style={styles.textSend}>SEND</Text>
                </TouchableOpacity>

                <View style={styles.line} />

                <TouchableOpacity style={styles.buttonRight} onPress={this.handleRecieve}>
                  <Image style={styles.iconRecieve} source={iconRecieve} />
                  <Text style={styles.textRecieve}>RECEIVE</Text>
                </TouchableOpacity>
              </View>

              <Transactions />
            </View>
          </SafeAreaView>

          <View style={styles.bottomPlaceForX} />

          {isVisibleNotify && <Notify />}

          {!modalVisible && <SpinnerPage />}
        </LinearGradient>
        <Modals navigation={navigation} />
      </Fragment>
    );
  }
}

const mapStateToProps = state => ({
  isModalSend: state.modals.isModalSend,
  isModalReceive: state.modals.isModalReceive,
  txDetails: state.modals.txDetails,
  transactions: state.wallet.transactions,
  isVisibleNotify: state.notify.isVisibleNotify,
  timeNotifyLast: state.notify.timeLast,
  isOpenDrawer: state.modals.isOpenDrawer,
  isModalSwitchContacts: state.modals.isModalSwitchContacts,
});

const mapDispatchToProps = dispatch => bindActionCreators(
  {
    setBalance: setBalanceAction,
    setAddress: setAddressAction,
    setTransactions: setTransactionsAction,
    setModalSend: setModalSendAction,
    setModalReceive: setModalReceiveAction,
    setModalTxDetails: setModalTxDetailsAction,

    setVisibleNotify: actionNotify.setVisibleNotify,
    setTimeNotify: actionNotify.setTimeNotify,

    getInfo: actionDashboard.getInfoAboutWallet,
    closeApp: closeAppAction,
    getContacts: actionDashboard.getContacts,
    showAlert: actionAlert.showAlert,
  },
  dispatch,
);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Dashboard);
