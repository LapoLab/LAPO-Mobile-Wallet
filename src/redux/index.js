import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import rootReducer from './rootReducer';
import rootSagas from './rootSaga';

const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
  blacklist: [
    'modals',
    'dashboard',
    'paperWords',
    'chooseContact',
    'send',
    'transactionsDetails',
    'alert',
    'pinsRecovery',
  ],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const sagaMiddleware = createSagaMiddleware();
// eslint-disable-next-line no-undef
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(sagaMiddleware)),
);
sagaMiddleware.run(rootSagas);

export const persistor = persistStore(store);
