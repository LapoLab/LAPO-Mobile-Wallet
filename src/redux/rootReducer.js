import { combineReducers } from 'redux';

import wallet from '../commons/reducers/wallet';
import pins from '../commons/reducers/pins';
import pinsRecovery from '../commons/reducers/pinsRecovery';
import modals from '../commons/reducers/modals';
import settings from '../commons/reducers/settings';
import notify from '../modules/Notify/reducers';
import paperConfirm from '../modules/PaperWordsConfirm/reducer';
import dashboard from '../modules/Dashboard/reducers';
import paperWords from '../modules/PaperWords/reducers';
import chooseContact from '../modules/ChooseContact/reducers';
import send from '../modules/Send/reducers';
import transactionsDetails from '../modules/TransactionDetails/reducer';
import alert from '../modules/Alert/reducers';

export default combineReducers({
  wallet,
  pins,
  pinsRecovery,
  notify,
  modals,
  settings,
  paperConfirm,
  dashboard,
  paperWords,
  chooseContact,
  send,
  transactionsDetails,
  alert,
});
