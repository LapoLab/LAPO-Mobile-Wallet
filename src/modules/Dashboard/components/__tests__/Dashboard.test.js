// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../redux';
import Dashboard from '../index';

describe('Dashboard page', () => {
  test('renders snapshots', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Dashboard />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
