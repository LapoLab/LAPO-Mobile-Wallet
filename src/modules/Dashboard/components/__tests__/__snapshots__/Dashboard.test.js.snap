// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard page renders snapshots 1`] = `
<BVLinearGradient
  colors={
    Array [
      -12483651,
      -16172276,
    ]
  }
  endPoint={
    Object {
      "x": 1.2,
      "y": 1.2,
    }
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  startPoint={
    Object {
      "x": -0.1,
      "y": -0.1,
    }
  }
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
        "zIndex": 0,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "elevation": 6,
          "flexDirection": "row",
          "height": 170,
          "justifyContent": "space-between",
          "marginTop": 30,
          "paddingHorizontal": 10,
          "shadowColor": "rgb(0,0,0)",
          "shadowOffset": Object {
            "height": 4,
            "width": 2,
          },
          "shadowOpacity": 0.34,
          "zIndex": 0,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "paddingBottom": 15,
            "paddingRight": 15,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/menuIcon.png",
            }
          }
          style={
            Object {
              "height": 18,
              "marginLeft": 8,
              "resizeMode": "contain",
              "width": 18,
            }
          }
        />
      </View>
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/logo-dashboard.png",
          }
        }
        style={
          Object {
            "height": 100,
            "marginRight": 15,
            "resizeMode": "contain",
            "width": 100,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 18,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/Activity.png",
            }
          }
          style={
            Object {
              "flex": 1,
              "marginRight": 8,
              "resizeMode": "contain",
              "width": 18,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "height": 20,
          "position": "absolute",
          "top": 134,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "Lato-Light",
            "fontSize": 14,
            "textAlign": "center",
          }
        }
      >
        blocks: 
        0
         - 0 %
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "borderRadius": 15,
            "elevation": 10,
            "height": 80,
            "left": 30,
            "paddingHorizontal": 15,
            "paddingVertical": 12,
            "position": "absolute",
            "right": 30,
            "shadowColor": "rgb(0,0,0)",
            "shadowOffset": Object {
              "height": 4,
              "width": 2,
            },
            "shadowOpacity": 0.34,
            "top": 156,
            "zIndex": 1,
          },
          Object {
            "backgroundColor": "#4183BD",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 2,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Lato-Bold",
              "fontSize": 26,
            }
          }
        >
          LAX
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Lato-Bold",
              "fontSize": 26,
            }
          }
        >
          0
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Lato-Light",
              "fontSize": 14,
            }
          }
        >
          $0.035
           
          / LAX
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Lato-Light",
              "fontSize": 14,
            }
          }
        >
          $0.00
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#EFF4FC",
          "flex": 3,
          "paddingBottom": 6,
          "zIndex": 0,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 150,
            "marginTop": 35,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "flex-start",
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/iconSend.png",
              }
            }
            style={
              Object {
                "height": 70,
                "marginTop": 25,
                "resizeMode": "contain",
                "width": 70,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#4183BD",
                "fontFamily": "Lato-Light",
                "fontSize": 16,
                "marginTop": 14,
              }
            }
          >
            SEND
          </Text>
        </View>
        <View
          style={
            Object {
              "borderRightColor": "#BCB8B8",
              "borderRightWidth": 1,
              "height": 100,
              "marginTop": 30,
              "width": 1,
            }
          }
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/iconRecieve.png",
              }
            }
            style={
              Object {
                "height": 70,
                "marginTop": 5,
                "resizeMode": "contain",
                "width": 70,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#7C8991",
                "fontFamily": "Lato-Light",
                "fontSize": 16,
                "marginTop": 10,
              }
            }
          >
            RECEIVE
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 15,
              "elevation": 10,
              "marginBottom": 44,
              "marginHorizontal": 28,
              "shadowColor": "rgb(0,0,0)",
              "shadowOffset": Object {
                "height": 4,
                "width": 2,
              },
              "shadowOpacity": 0.34,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4183BD",
                "borderTopLeftRadius": 15,
                "borderTopRightRadius": 15,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-around",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Lato-Regular",
                    "fontSize": 18,
                  }
                }
              >
                ALL
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Lato-Light",
                    "fontSize": 18,
                    "opacity": 0.7,
                  }
                }
              >
                INCOME
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "Lato-Light",
                    "fontSize": 18,
                    "opacity": 0.7,
                  }
                }
              >
                OUTCOME
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "white",
                "borderBottomLeftRadius": 15,
                "borderBottomRightRadius": 15,
                "minHeight": 116,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "paddingVertical": 25,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontFamily": "Lato-Light",
                    "marginBottom": 12,
                  }
                }
              >
                no transactions yet
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
  <View
    style={
      Object {
        "backgroundColor": "#EFF4FC",
        "height": 0,
      }
    }
  />
</BVLinearGradient>
`;
